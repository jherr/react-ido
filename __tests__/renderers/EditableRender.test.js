import React from 'react';
import TestRenderer from 'react-test-renderer';

import {
  IdoSurface,
  Palette,
  EditableRender,
  InlineEditable,
} from '../../src';

const Field = ({ name, value }) => (
  <div>
    {`${name}: ${value}`}
  </div>
);

const EditableField = InlineEditable({
  value: {
    name: 'Value',
  },
})(Field);

const config = {
  widgets: [
    {
      type: 'field',
      name: 'Field',
      icon: 'list',
      component: Field,
      editComponent: EditableField,
      defaultValues: {
        name: 'Foo',
        value: 'Bar'
      }
    },
  ],
};

const elements = [
  [
    {
      type: 'field',
      id: 2,
      data: {
        name: 'Price',
        value: '$20.00',
      },
      width: 6,
    },
  ],
];

describe('EditableRenderer', () => {
  it('should render', () => {
    expect(
      TestRenderer.create(
        <IdoSurface
          config={config}
          elements={elements}
          onChange={() => {}}
        >
          <Palette />
          <EditableRender />
        </IdoSurface>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
