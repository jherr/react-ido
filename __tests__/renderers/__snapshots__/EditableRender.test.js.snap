// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableRenderer should render 1`] = `
Array [
  <div>
    <div>
      <div
        className="css-1i9po67"
      >
        <i
          className="oi oi-list"
          style={
            Object {
              "paddingRight": 5,
            }
          }
        />
        Field
      </div>
    </div>
  </div>,
  <div>
    <div>
      <div
        className={null}
        style={
          Object {
            "border": "1px dashed #ccc",
            "borderRadius": 5,
            "margin": 2,
            "textAlign": "center",
          }
        }
      >
        <i
          className="oi oi-target"
          style={
            Object {
              "color": "#ccc",
              "height": 30,
              "opacity": 0,
              "width": 30,
            }
          }
        />
      </div>
    </div>
    <div
      className="css-chr1eq"
      style={
        Object {
          "gridTemplateColumns": "repeat(1, 1fr)",
        }
      }
    >
      <div
        className="rounded"
        style={
          Object {
            "padding": 2,
          }
        }
      >
        <div>
          <div
            className="css-zpu786"
          >
            <div>
              <button
                className="css-7yzxl9"
                onClick={[Function]}
              >
                <i
                  className="oi oi-circle-x"
                  style={
                    Object {
                      "color": "white",
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </button>
            </div>
            <div
              className="css-1paq4b3"
            >
              Field
            </div>
            <div>
              <button
                className="css-7yzxl9"
                onClick={[Function]}
              >
                <i
                  className="oi oi-cog"
                  style={
                    Object {
                      "color": "white",
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
              </button>
            </div>
            <div
              className={null}
              style={
                Object {
                  "alignItems": "center",
                  "border": "1px dashed #ccc",
                  "borderRadius": 5,
                  "display": "flex",
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                  "margin": 2,
                  "padding": "2px 5px 2px 5px",
                  "textAlign": "center",
                }
              }
            >
              <i
                className="oi oi-target"
                style={
                  Object {
                    "color": "#ccc",
                    "height": 30,
                    "opacity": 0,
                    "width": 30,
                  }
                }
              />
            </div>
            <div
              className="css-xrwow"
            >
              <div>
                Price: $20.00
              </div>
            </div>
            <div
              className={null}
              style={
                Object {
                  "alignItems": "center",
                  "border": "1px dashed #ccc",
                  "borderRadius": 5,
                  "display": "flex",
                  "flexDirection": "column",
                  "justifyContent": "space-around",
                  "margin": 2,
                  "padding": "2px 5px 2px 5px",
                  "textAlign": "center",
                }
              }
            >
              <i
                className="oi oi-target"
                style={
                  Object {
                    "color": "#ccc",
                    "height": 30,
                    "opacity": 0,
                    "width": 30,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className={null}
      style={
        Object {
          "border": "1px dashed #ccc",
          "borderRadius": 5,
          "margin": 2,
          "textAlign": "center",
        }
      }
    >
      <i
        className="oi oi-target"
        style={
          Object {
            "color": "#ccc",
            "height": 30,
            "opacity": 0,
            "width": 30,
          }
        }
      />
    </div>
  </div>,
]
`;
